tests/cnf/core/network/sriov/tests/qinq.go

```diff
@@ -544,6 +578,152 @@ var _ = Describe("QinQ", Ordered, Label(tsparams.LabelQinQTestCases), ContinueOn
 		})
 	})
 
+	Context("nmstate", func() {
+		const configureNMStatePolicyName = "configurevfs"
+		var (
+			sriovInterfacesUnderTest []string
+			workerNodeList           []*nodes.Builder
+		)
+		BeforeAll(func() {
+			By("Remove non-nmstate sriov configurations")
+			removeSriovConfigurations()
+			By("Waiting until cluster MCP and SR-IOV are stable")
+			err = netenv.WaitForSriovAndMCPStable(
+				APIClient, tsparams.MCOWaitTimeout, time.Minute, NetConfig.CnfMcpLabel, NetConfig.SriovOperatorNamespace)
+			Expect(err).ToNot(HaveOccurred(), "Failed cluster is not stable")
+
+			By("Creating a new instance of NMstate instance")
+			err = netnmstate.CreateNewNMStateAndWaitUntilItsRunning(7 * time.Minute)
+			Expect(err).ToNot(HaveOccurred(), "Failed to create NMState instance")
+
+			By("Validating SR-IOV interfaces")
+			workerNodeList, err = nodes.List(APIClient,
+				metav1.ListOptions{LabelSelector: labels.Set(NetConfig.WorkerLabelMap).String()})
+			Expect(err).ToNot(HaveOccurred(), "Failed to discover worker nodes")
+
+			Expect(sriovenv.ValidateSriovInterfaces(workerNodeList, 2)).ToNot(HaveOccurred(),
+				"Failed to get required SR-IOV interfaces")
+			sriovInterfacesUnderTest, err = NetConfig.GetSriovInterfaces(2)
+			Expect(err).ToNot(HaveOccurred(), "Failed to retrieve SR-IOV interfaces for testing")
```

Comment:
I guees, you did it in the prious tests, right?  could you take values from there?

Reply:
having issues passing the value into context. Lets talk this evening